<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>SpatialGEE Integration Tutorial</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">SpatialGEE Integration Tutorial</h1>



<div id="introduction" class="section level1">
<h1>1. Introduction</h1>
<p>This tutorial demonstrates how to use <strong>SpatialGEE</strong> for
spatial co-profiling integration of ATAC and RNA data. We will:</p>
<ul>
<li><p>Perform differential expression analysis using
<code>run_gee_gst()</code> and <code>run_wilcoxon()</code>.</p></li>
<li><p>Use IMIX for integrative analysis based on P-values.</p></li>
</ul>
</div>
<div id="installation-and-loading" class="section level1">
<h1>2. Installation and Loading</h1>
<p>To use the <strong>SpatialGEE</strong> package, install it and load
it into R session:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;devtools&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;yishan03/SpatialGEE&quot;</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(SpatialGEE)</span></code></pre></div>
</div>
<div id="example-dataset" class="section level1">
<h1>3. Example Dataset</h1>
<p>The package includes an example dataset,
<code>coprofile_example_data</code>, based on a subset of the processed
spatial epigenome-transcriptome mouse brain co-profiling data (Zhang et
al., 2023).</p>
<div id="dataset-description" class="section level2">
<h2>3.1. Dataset Description</h2>
<ul>
<li><p>The dataset contains spatial co-profiling data (ATAC &amp; RNA),
including metadata and gene-level measurements across 9,215
cells.</p></li>
<li><p>Metadata columns include:</p>
<ul>
<li><code>Barcodes</code>: Cell barcodes.</li>
<li><code>x</code>, <code>y</code>: Spatial coordinates.</li>
<li><code>Pathology.Annotations</code>: Pathology labels
(<code>&quot;non-Corpus callosum&quot;</code> and
<code>&quot;Corpus callosum&quot;</code>).</li>
</ul></li>
<li><p>Gene expression and accessibility data include 100 selected
genes: <code>Gabbr2</code>, <code>Pde7b</code>, …,
<code>Itga8</code>.</p></li>
</ul>
</div>
<div id="dataset-example" class="section level2">
<h2>3.2. Dataset Example</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(coprofile_example_data)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>atac <span class="ot">&lt;-</span> coprofile_example_data<span class="sc">$</span>ATAC</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>rna <span class="ot">&lt;-</span> coprofile_example_data<span class="sc">$</span>RNA</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">head</span>(coprofile_example_data<span class="sc">$</span>ATAC) <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;           Barcodes  x y Pathology.Annotations   Gabbr2    Pde7b    Rims1</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; 1 AAACATCGAACTCACC 31 2   non-Corpus callosum 1.571669 1.268047 1.219150</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; 2 AAACATCGAAGAGATC 30 2   non-Corpus callosum 1.523837 1.406939 1.221459</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; 3 AAACATCGAAGGACAC 29 2   non-Corpus callosum 1.516391 1.580516 1.247479</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; 4 AAACATCGAATCCGTC 28 2   non-Corpus callosum 1.402581 1.591711 1.249481</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; 5 AAACATCGAATGTTGC 27 2   non-Corpus callosum 1.450364 1.572980 1.272308</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; 6 AAACATCGACACAGAA 82 2   non-Corpus callosum 1.542466 1.032041 1.039183</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">head</span>(coprofile_example_data<span class="sc">$</span>RNA) <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt;           Barcodes  x y Pathology.Annotations Gabbr2 Pde7b Rims1</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt; 1 AAACATCGAACTCACC 31 2   non-Corpus callosum      0     0     0</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; 2 AAACATCGAAGAGATC 30 2   non-Corpus callosum      0     0     0</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 3 AAACATCGAAGGACAC 29 2   non-Corpus callosum      0     0     3</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 4 AAACATCGAATCCGTC 28 2   non-Corpus callosum      0     0     1</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 5 AAACATCGAATGTTGC 27 2   non-Corpus callosum      1     0     7</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 6 AAACATCGACACAGAA 82 2   non-Corpus callosum      0     0     0</span></span></code></pre></div>
</div>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>4. Differential Expression Analysis</h1>
<div id="continious-data-type" class="section level2">
<h2>4.1. Continious data type</h2>
<p>We use <code>run_wilcoxon()</code> to contentious data types, such as
normalized ATAC.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>atac_DE_res <span class="ot">&lt;-</span> <span class="fu">run_wilcoxon</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  atac, </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">compare_levels =</span> <span class="fu">c</span>(<span class="st">&quot;non-Corpus callosum&quot;</span>, <span class="st">&quot;Corpus callosum&quot;</span>))</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">head</span>(atac_DE_res)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;       gene       p_value</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; 1   Gabbr2  7.429736e-32</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 2    Pde7b  1.385373e-89</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; 3    Rims1 8.215698e-167</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; 4   Snap25 8.054409e-202</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; 5    Il12a  1.722868e-01</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; 6 Cacna2d3  9.952009e-43</span></span></code></pre></div>
</div>
<div id="count-data-type" class="section level2">
<h2>4.2. Count data type</h2>
<p>We use <code>run_gee_gst()</code> to count data types, such as RNA
counts.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>rna_DE_res <span class="ot">&lt;-</span> <span class="fu">run_gee_gst</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  rna, </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">compare_levels =</span> <span class="fu">c</span>(<span class="st">&quot;non-Corpus callosum&quot;</span>, <span class="st">&quot;Corpus callosum&quot;</span>))</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="fu">head</span>(rna_DE_res)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt;       gene      p_value</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; 1   Gabbr2 0.0005489216</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; 2    Pde7b 0.0125448984</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; 3    Rims1 0.5848623358</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; 4   Snap25 0.0006652406</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; 5    Il12a 0.0004134127</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; 6 Cacna2d3 0.0020262216</span></span></code></pre></div>
</div>
<div id="combine-atac-rna-p-values" class="section level2">
<h2>4.3. Combine ATAC &amp; RNA P-values</h2>
<p>We combine the results from ATAC and RNA.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>merged_pvalue <span class="ot">&lt;-</span> <span class="fu">merge</span>(atac_DE_res, rna_DE_res, <span class="at">by =</span> <span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span> <span class="fu">c</span>(<span class="st">&quot;atac_pvalue&quot;</span>, <span class="st">&quot;rna_pvalue&quot;</span>), <span class="fu">starts_with</span>(<span class="st">&quot;p_value&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="fu">head</span>(merged_pvalue)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt;          atac_pvalue   rna_pvalue</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; Acot11  1.381807e-78 4.918471e-01</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt; Ak4     5.703768e-04 2.031796e-02</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt; Ak5    3.708405e-119 1.695025e-03</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt; Alas1   3.372929e-14 8.999031e-01</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt; Ank     3.119800e-24 5.372984e-01</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co">#&gt; Apod    2.115761e-23 1.645404e-09</span></span></code></pre></div>
</div>
</div>
<div id="run-imix-for-atac-rna-integration" class="section level1">
<h1>5. Run IMIX for ATAC-RNA Integration</h1>
<p>We run IMIX for integration and retrieve significant genes with FDR
control.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>integration_res <span class="ot">&lt;-</span> IMIX<span class="sc">::</span><span class="fu">IMIX</span>(<span class="at">data_input =</span> merged_pvalue)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#&gt; Assign initial values</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt; number of iterations= 39 </span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt; number of iterations= 93 </span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; Start IMIX-ind procedure!</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt; Successfully Done!</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt; Start IMIX-cor-twostep procedure!</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; Successfully Done!</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#&gt; Start IMIX-cor model procedure!</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="co">#&gt; Successfully Done!</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="co">#&gt; Start IMIX-cor-restrict procedure!</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co">#&gt; Successfully Done!</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#&gt; Warning: IMIX_cor_restrict did not converge. </span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#&gt; Assign IMIX_ind results to IMIX_cor_restrict. </span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co">#&gt; The AIC/BIC values for IMIX_cor_restrict are not reliable and should not be used!</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#&gt; Start Model Selection</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co">#&gt; Start Label Sorting</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co">#&gt; Start Adaptive FDR Control</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="co">#&gt; Finished!</span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="fu">head</span>(integration_res<span class="sc">$</span>significant_genes_with_FDRcontrol)</span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="co">#&gt;        localFDR class_withoutFDRcontrol class_FDRcontrol</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="co">#&gt; Ak5           0                       4                4</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a><span class="co">#&gt; Arfip1        0                       4                4</span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a><span class="co">#&gt; Atp2b2        0                       4                4</span></span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a><span class="co">#&gt; Cdc5l         0                       4                4</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a><span class="co">#&gt; Cpne5         0                       4                4</span></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a><span class="co">#&gt; Dgkb          0                       4                4</span></span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>Wang, Y., &amp; Wei, P. (2025). A Mixture Model Approach for
Integrating Spatial Transcriptomics and Epigenomics. Unpublished
manuscript.</p>
<p>Wang, Z., &amp; Wei, P. (2020). IMIX: A multivariate mixture model
approach to association analysis through multi-omics data integration.
Bioinformatics, 36(22–23), 5439–5447.</p>
<p>Zhang, D., Deng, Y., Kukanja, P., Agirre, E., Bartosovic, M., Dong,
M., Ma, C., Ma, S., Su, G., Bao, S., Liu, Y., Xiao, Y., Rosoklija, G.
B., Dwork, A. J., Mann, J. J., Leong, K. W., Boldrini, M., … Fan, R.
(2023). Spatial epigenome–transcriptome co-profiling of mammalian
tissues. Nature, 616(7955), 113–122.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
